management/src/test/java/org/ehcache/management/registry/DefaultManagementRegistryTest.java

```diff
@@ -121,18 +133,185 @@ public void testCanGetStats() {
+        .on(context1)
+        .on(context2)
+        .build()
+        .execute();
 
-    assertThat(allCounters, hasSize(2));
-    assertThat(allCounters.get(0), hasSize(1));
-    assertThat(allCounters.get(1), hasSize(1));
-    assertThat(allCounters.get(0).iterator().next().getValue(), equalTo(2L));
-    assertThat(allCounters.get(1).iterator().next().getValue(), equalTo(3L));
+    assertThat(allCounters.size(), equalTo(2));
+    assertThat(allCounters.get(context1).getStatistics(), hasSize(1));
+    assertThat(allCounters.get(context2).getStatistics(), hasSize(1));
+    assertThat(allCounters.get(context1).getStatistic(0, Counter.class).getValue(), equalTo(2L));
+    assertThat(allCounters.get(context2).getStatistic(0, Counter.class).getValue(), equalTo(3L));
+
+    cacheManager1.close();
+  }
+
+  @Test
+  public void testCanGetStatsSinceTime() throws InterruptedException {
+    CacheConfiguration<Long, String> cacheConfiguration = CacheConfigurationBuilder.newCacheConfigurationBuilder()
+        .withResourcePools(ResourcePoolsBuilder.newResourcePoolsBuilder().heap(10, EntryUnit.ENTRIES).build())
+        .buildConfig(Long.class, String.class);
+
+    ManagementRegistry managementRegistry = new DefaultManagementRegistry(new DefaultManagementRegistryConfiguration()
+        .addConfiguration(new EhcacheStatisticsProviderConfiguration(5000, TimeUnit.MILLISECONDS, 100, 1, TimeUnit.SECONDS, 30, TimeUnit.SECONDS))
+        .setCacheManagerAlias("myCM"));
+
+    CacheManager cacheManager1 = CacheManagerBuilder.newCacheManagerBuilder()
+        .withCache("aCache1", cacheConfiguration)
+        .using(managementRegistry)
+        .build(true);
+
+    Map<String, String> context = new HashMap<String, String>();
+    context.put("cacheManagerName", "myCM");
+    context.put("cacheName", "aCache1");
+
+    StatisticQuery.Builder builder = managementRegistry.withCapability("StatisticsCapability")
+        .queryStatistics("AllCachePutCount", "AllCachePutRate", "AllCachePutLatencyAverage")
+        .on(context);
+
+    ContextualStatistics statistics;
+    CounterHistory putCount;
+    RateHistory putRate;
+    AverageHistory putAverageLatency;
+    long timestamp;
+
+    // ------
+    // first call to trigger stat collection within ehcache stat framework
+    // ------
+
+    builder.build().execute();
+
+    // ------
+    // 3 puts and we wait more than 1 second (history frequency) to be sure the scheduler thread has computed a new stat in the history
+    // ------
+
+    cacheManager1.getCache("aCache1", Long.class, String.class).put(1L, "1");
+    cacheManager1.getCache("aCache1", Long.class, String.class).put(2L, "2");
+    cacheManager1.getCache("aCache1", Long.class, String.class).put(2L, "2");
+
+    do {
+      Thread.sleep(100);
+      statistics = builder.build().execute().get(context);
+      putCount = statistics.getStatistic(0, CounterHistory.class);
+    } while (putCount.getValue().size() < 1);
```

Comment:
There should be a "safety exit" from this loop, in case a bug makes it loop forever. For readability, this should also be extracted into a "assert*" method.